### ISSUE TITLE
Amend documentation to reflect that `meson install` is not run

### ISSUE THREAD

**Issue by dnicolodi** [mesonbuild/meson-python#473]
Amend the documentation to reflect the fact that we don't actually run meson install but just re-implement it. This affects two Meson features: `install_rpath` support, and running scripts registered with `meson.add_install_script()`. Both things have no effect when building a package with `meson-python`.

**Comment by dnicolodi** [comment]
Also explicitly list the `meson install` options recognized by `meson-python` in https://meson-python.readthedocs.io/en/latest/reference/pyproject-settings.html#cmdoption-arg-tool.meson-python.args.install and https://meson-python.readthedocs.io/en/latest/reference/config-settings.html#cmdoption-arg-install-args or find a way to do not have to repeat the content twice.

Most of the `meson install` options not handled by `meson-python` aren't relevant, but it's possible `--strip` may be or some new option gets added to `meson install` that we must start handling.

**Comment by FFY00** [comment]
I ran into this today, where I explicitly needed to perform some artifact fixup that was not possible any other sane way in meson other than using `meson.add_install_script()`.

**Comment by dnicolodi** [comment]
@FFY00 AFAIK the only feature offered to a script executed as a `meson.add_install_script()` vs a script executed as a `custom_target()` is the knowledge of the installation location via environment variables like `MESON_INSTALL_PREFIX` and `MESON_INSTALL_DESTDIR_PREFIX`. The installation path is hard to define for packages installed as wheels. Therefore, I am not able to imagine what requirement makes an installation script the right solution. Can you explain your use case in more detail?

**Comment by eli-schwartz** [comment]
Stuff like auditwheel repair, maybe?

**Comment by FFY00** [comment]
> Stuff like auditwheel repair, maybe?

Yes, exactly. In my particular case, I needed to perform a small edit to the binary format data of certain artifacts.

This is not possible in Meson AFAIK, as you can't pass a build target as the input for a custom target,

**Comment by dnicolodi** [comment]
The Meson documentation is not very clear on this, but you can pass the object returned by `build_target()` as input to a `custom_target()`.
